charts/kong/templates/cluster-certificate.yaml

```
@@ -0,0 +1,21 @@
+{{- if and ( .Capabilities.APIVersions.Has "cert-manager.io/v1" ) .Values.certificate.cluster.enabled }}
+# Self Signed Kong Cluster Certificate
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: kong-cluster
```

Generated resource names should be such that they cannot conflict with other releases' resources. We don't require a dedicated namespace per release.

We also add the namespace explicitly to include it when rendered through `helm template` instead of `helm install`.

```suggestion
  name: {{ include "kong.fullname" . }}-kong-cluster
  namespace:  {{ template "kong.namespace" . }}
```